### DISCUSSION TITLE
Add TSX Support?

### DISCUSSION THREAD

**Discussion by danjguzman** [brackets-cont/brackets#247] [Feature Request]
Currently the only real **TypeScript-JSX** (React-TS) support in Brackets for proper **syntax highlighting** is an _old_ extension that is not maintained. The problem is that it hasn't been updated and using **?. Optional Chaining** (obj.key?.x), it removes the syntax highlighting for any code after using Optional Chaining creating total havoc.

Without Optional Chaining. Notice **let object = {};** has proper syntax highlighting.

![Screen Shot 2023-01-08 at 5 58 11 PM](https://user-images.githubusercontent.com/33531668/211230805-366e060d-b1aa-4726-b299-0349cc3a4937.png)

With Optional Chaining. Notice **let object = {};** has the wrong syntax highlighting, and all code below optional chaining will be white.

![Screen Shot 2023-01-08 at 5 58 00 PM](https://user-images.githubusercontent.com/33531668/211230803-31377743-014d-4d16-90a7-4c7344e6fd69.png)

This is also, sometimes, an issue with Template Literals, where all JSX/HTML code and any props will just be orange.

This is also an issue when using JSX for syntax highlighting. It would be great to get this updated like other editors that properly highlight syntax.

Thanks.

**Comment by abose** [comment]
@danjguzman Can you see if the syntax issue is resolved in https://phcode.dev ?

For the typescript extension support, we are adding better language intelligence directly in core without any extensions. Please follow this issue for details: https://github.com/phcode-dev/phoenix/issues/27

**Comment by danjguzman** [comment]
I tested it on PHCode and it appears the issue would be resolved there. Any following code no longer loses its syntax highlight colors.

The only issue I do see in the PHCode editor is that the syntax colors are off, there is this additional plumb looking color that doesn't match the lavender "variable" color. In the original Brackets Dark Theme, all of them would match the same color to keep them all consistent.

![Screen Shot 2023-01-10 at 1 47 45 PM](https://user-images.githubusercontent.com/33531668/211669536-0e224c35-d18c-490e-86b8-7bbb187236fb.png)

Overall syntax highlighting is **much improved**. I would prefer to use this as a **desktop**, not _browser_ app though.
